raft/src/test/java/org/apache/kafka/raft/LeaderStateTest.java

```diff
@@ -188,13 +278,13 @@ public void testNonMonotonicHighWatermarkUpdate() {
         int node1 = 1;
         LeaderState<?> state = newLeaderState(mkSet(localId, node1), 0L);
         state.updateLocalState(time.milliseconds(), new LogOffsetMetadata(10L));
-        state.updateReplicaState(node1, time.milliseconds(), new LogOffsetMetadata(10L));
+        state.updateReplicaState(node1, time.milliseconds(), new LogOffsetMetadata(10L), 11L);
         assertEquals(Optional.of(new LogOffsetMetadata(10L)), state.highWatermark());
 
         // Follower crashes and disk is lost. It fetches an earlier offset to rebuild state.
         // The leader will report an error in the logs, but will not let the high watermark rewind
-        assertFalse(state.updateReplicaState(node1, time.milliseconds(), new LogOffsetMetadata(5L)));
-        assertEquals(5L, state.getVoterEndOffsets().get(node1));
+        assertFalse(state.updateReplicaState(node1, time.milliseconds(), new LogOffsetMetadata(5L), 6L));
```

Comment:
Why does the leader end offset go backwards here? Previously it was at 11.

Reply:
Accidental update. Will fix.